include/posix/pthread.h

```diff
@@ -7,39 +7,42 @@
 #ifndef __PTHREAD_H__
 #define __PTHREAD_H__
 
-#ifdef CONFIG_NEWLIB_LIBC
+#include <kernel.h>
 #include <time.h>
-#else
-/* This should probably live somewhere else but Zephyr doesn't
- * currently have a stdc layer to provide it
+#include "sys/types.h"
+#include "posix_sched.h"
+#include "unistd.h"
+
+struct posix_thread {
+	struct k_thread thread;
+	void *retval;
+};
+
+/* Pthread cancellation */
+#define PTHREAD_CANCEL_POS     0
+/* Non Essential Thread */
+#define PTHREAD_CANCELABLE     (0 << PTHREAD_CANCEL_POS)
+/* Essential Thread */
+#define PTHREAD_NON_CANCELABLE (1 << PTHREAD_CANCEL_POS)
+#define THREAD_INIT_FLAGS       PTHREAD_CANCELABLE
```

Comment:
Why is this prefixed with `THREAD_`?  If someone includes `pthread.h`, this will pollute their namespace with things that are not POSIX-specific. There doesn't seem to exist any `THREAD_INIT_FLAGS` in Linux as well, so I don't know why is this defined in the header file.